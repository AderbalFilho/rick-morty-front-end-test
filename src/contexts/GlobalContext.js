import React, { createContext, useState } from 'react';

export const GlobalContext = createContext();

export const GlobalStorage = ({ children }) => {
  const [searchTerm, setSearchTerm] = useState('');
  const [characters, setCharacters] = useState([]);
  const [isLoading /*, setIsLoading*/] = useState(false);

  function handleSearch() {
    // Do search
    setCharacters([
      /* result */
    ]);
  }

  return (
    <GlobalContext.Provider
      value={{ searchTerm, setSearchTerm, characters, handleSearch, isLoading }}
    >
      {children}
    </GlobalContext.Provider>
  );
};
