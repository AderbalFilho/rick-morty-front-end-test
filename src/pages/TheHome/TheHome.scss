.the-home {
  background: var(--color-black);
  min-height: 100vh;
  min-width: 100%;
}

.the-home--blur {
  filter: var(--filter-blur);
  overflow: hidden;
  position: fixed;

  @media (min-width: 768px) {
    filter: var(--filter-blur-desktop);
  }
}

.the-home--sheet-open {
  overflow: hidden;
  position: fixed;
}

.the-home__img-background {
  background-image: url('/assets/imgs/background.png');
  background-position: top;
  background-repeat: no-repeat;
  min-height: 900px;
}

.the-home__cards {
  display: grid;
  gap: 24px;
  grid-template-columns: repeat(auto-fit, 224px);
  justify-content: center;
  margin: 0 auto;
  max-width: 968px;
  padding: 47px 0;

  @media (min-width: 768px) {
    padding: 101px 0;
  }
}

.the-home__pagination-section {
  display: flex;
  justify-content: center;
  padding-bottom: 73px;

  @media (min-width: 768px) {
    padding-bottom: 203px;
  }
}
