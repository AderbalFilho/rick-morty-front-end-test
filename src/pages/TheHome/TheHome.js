import React, { useContext } from 'react';
import classNames from 'classnames/bind';

import SearchSection from '@components/SearchSection';
import TheLoading from '@components/TheLoading';
import CharacterCard from '@components/CharacterCard';
import { GlobalContext } from '@contexts/GlobalContext';

import './TheHome.scss';

function TheHome() {
  const { characters, isLoading } = useContext(GlobalContext);
  const className = classNames('the-home', isLoading && ' the-home--blur');

  return (
    <>
      {isLoading && <TheLoading />}
      <div className={className}>
        <div className="the-home__img-background">
          <SearchSection />
          <section className="the-home__cards">
            {characters.map((character) => (
              <CharacterCard key={character.id} character={character} />
            ))}
          </section>
        </div>
      </div>
    </>
  );
}

export default TheHome;
