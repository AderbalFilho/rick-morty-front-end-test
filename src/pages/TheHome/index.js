import TheHome from './TheHome';

export default TheHome;
