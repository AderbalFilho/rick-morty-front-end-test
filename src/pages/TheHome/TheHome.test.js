import React from 'react';
import TestRenderer from 'react-test-renderer';

import TheHome from './TheHome';
import { GlobalStorage } from '@contexts/GlobalContext';

describe('TheHome', () => {
  it('matches snapshot', () => {
    const testRenderer = TestRenderer.create(
      <GlobalStorage>
        <TheHome />
      </GlobalStorage>
    );
    const jsonComponent = testRenderer.toJSON();

    expect(jsonComponent).toMatchSnapshot();
  });
});
