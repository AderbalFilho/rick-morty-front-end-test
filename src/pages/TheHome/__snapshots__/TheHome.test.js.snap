// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TheHome matches snapshot 1`] = `
<div
  className="the-home"
>
  <div
    className="the-home__img-background"
  >
    <section
      className="search-section"
    >
      <img
        alt="Rick and Morty logo"
        className="search-section__logo"
        src=""
      />
      <div
        className="search-section__input-container"
      >
        <input
          className="g-input"
          onChange={[Function]}
          placeholder="Search characters"
          type="text"
          value=""
        />
        <button
          className="g-button search-section__button"
          onClick={[Function]}
        >
          Search
        </button>
      </div>
    </section>
    <section
      className="the-home__cards"
    />
  </div>
</div>
`;
